### PR TITLE
fix: removed the possibility of concurrent webauth transactions to handle continuation misuse

### PR DESCRIPTION
**Pull request by desusai7**
### 📋 Changes

* Updated WebAuth to cancel any newly triggered WebAuth transactions when there exists an active WebAuth transaction already i.e ensuring that only single WebAuth transaction is active at any point of time.
* The above change would fix the issues in some special cases of race condition where continuations are misused by resuming twice.
* Updated unit tests for `login` & `logout` in WebAuthSpec to clear `TransactionStore` before each run to start clean and not get failed by the un-cleared transaction in memory by previous test runs.

### 📎 References

[auth0/Auth0.swift#827](https://github.com/auth0/Auth0.swift/issues/827)

### Checklist

- [x] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### 🎯 Testing

Tested by trying to start the below defined two flows at exact same time via both callbacks & async/await approach and found that the code is working as per expectations **by cancelling the newly triggering web auth flow and ensuring the existing web auth flow executes completely.** 

1. WebAuth flow with `google-oauth2` connection using `ASWebAuthenticationSession`
2. WebAuth flow with `Universal Login` using `SafariViewController`

### CODE REVIEW

File: Auth0/WebAuthError.swift [hunk 1]
```diff
@@ -79,6 +80,7 @@ extension WebAuthError {
         switch self.code {
         case .noBundleIdentifier: return "Unable to retrieve the bundle identifier from Bundle.main.bundleIdentifier,"
             + " or it could not be used to build a valid URL."
+        case .transactionActiveAlready: return "Failed to start this transaction, as there is an active transaction at the moment"
```
Comment (Widcket) [inline Auth0/WebAuthError.swift:83]:
```suggestion
        case .transactionActiveAlready: return "Failed to start this transaction, as there is an active transaction at the moment."
```

File: Auth0/WebAuthError.swift [hunk 2]
```diff
@@ -6,6 +6,7 @@ public struct WebAuthError: Auth0Error {
 
     enum Code: Equatable {
         case noBundleIdentifier
+        case transactionActiveAlready
```
Comment (Widcket) [inline Auth0/WebAuthError.swift:9]: There are tests for `WebAuthError`, those should be updated as well: https://github.com/auth0/Auth0.swift/blob/master/Auth0Tests/WebAuthErrorSpec.swift

Comment (desusai7) [reply Auth0/WebAuthError.swift:9]: Thanks for pointing it out, included tests for it now.

File: Auth0/WebAuthError.swift (hunk 1 above)
Comment (desusai7) [reply Auth0/WebAuthError.swift]: Updated the text description for the WebAuthError `.transactionActiveAlready` as per description

File: Auth0/WebAuthError.swift (hunk 2 above)
Comment (Widcket) [reply Auth0/WebAuthError.swift:9]: There's also a test for the error code: https://github.com/auth0/Auth0.swift/blob/master/Auth0Tests/WebAuthErrorSpec.swift#L27

**Review by Widcket — APPROVED** [review]

**Pull request merged**